src/docker/composeappengine.cc

```diff
@@ -42,11 +42,7 @@ bool ComposeAppEngine::fetch(const App& app) {
 
     result = true;
   } catch (const std::exception& exc) {
-    LOG_WARNING << "Failed to get/set App state, fallback to forced fetch: " << exc.what();
-
-    if (download(app) && verify(app) && pullImages(app)) {
-      result = true;
-    }
```

Comment:
not sure i follow why this logic would change?

Reply:
The overall logic hasn't changed as result of the AppState::setState() implementation change, it doesn't throw exception now, so the net effect of both implementations is the same.

This "state" logic of composeappengine is really weird I wish I hadn't implemented it at all, it's just optimizes a number fetch tries but makes code more complicated :).